core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java

```diff
@@ -69,10 +72,63 @@ public boolean print(byte[] content, JsonNode configuration) {
       return true;
     }
   },
+  LINE_PRINTER("host", "queue") {
+
+    @Override
+    public boolean print(byte[] content, JsonNode configuration, User user) {
+      String host = configuration.get("host").asText();
+      String queue = configuration.get("queue").asText();
+      // We need a three digit job id.
+      int id = Arrays.hashCode(content);
+      if (id < 0) {
+        id = -id;
+      }
+      id = id % 1000;
+      try (Socket socket = new Socket(host, 515);
+          OutputStream output = socket.getOutputStream();
+          InputStream input = socket.getInputStream()) {
+        output.write(2);
+        output.write(queue.getBytes(StandardCharsets.US_ASCII));
+        output.write(10);
+        output.flush();
+        if (input.read() != 0) {
+          return false;
+        }
+        byte[] control = String.format("Hmiso\nP%s\nvdfA%03dmiso\nN(gui)\nUdfA%03dmiso\n", user.getLoginName(), id, id)
+            .getBytes(StandardCharsets.US_ASCII);
+        output.write(String.format("\002%d cfA%03dmiso\n", control.length, id).getBytes(StandardCharsets.US_ASCII));
+        output.flush();
+        if (input.read() != 0) {
+          return false;
+        }
+        output.write(control);
+        output.write(0);
+        if (input.read() != 0) {
+          return false;
+        }
+        output.write(String.format("\003%d dfA%03dmiso\n", content.length, id).getBytes(StandardCharsets.US_ASCII));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') %n should be used in place of \n to produce the platform-specific line separator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)